app/config/FeatureSwitchConfig.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import features.{Feature, FeatureStatus}
+import features.Feature.Feature
+import features.FeatureStatus.FeatureStatus
+import play.api.Configuration
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class FeatureSwitchConfig @Inject()(configuration: Configuration) {
+
+  private def loadConfig(key: String): String =
+    configuration.getOptional[String](key).getOrElse(throw new Exception(s"Missing configuration key: $key"))
+
+  private def feature2Key(feature: Feature): String = s"microservice.services.features.$feature"
+
+  private def str2FeatureStatus(str: String): FeatureStatus = FeatureStatus.withName(str)
+
+  lazy val defaultFeatureStatus: features.FeatureStatus.Value =
+    FeatureStatus.withName(loadConfig(feature2Key(Feature.default)))
+
+  private def getFeatureStatusFromProperties(feature: Feature): Option[FeatureStatus] = {
+    println(feature)
+    println(feature2Key(feature))
```

Comment:
`println`?

Reply:
Whoops! There are no pages that use the flag so I was using this to test. Removed :+1: